lib/portlayer/storage/volume_cache.go

```diff
@@ -232,3 +234,22 @@ func volumeInUse(ID string) error {
 
 	return nil
 }
+
+func (v *VolumeLookupCache) StatPath(op trace.Operation, deviceId string, target string) (*compute.FileStat, error) {
+	v.vlcLock.Lock()
+	defer v.vlcLock.Unlock()
+	// FIXME: If we change the structure of volume cache, we can no longer get volume store from a deviceID.
```

Comment:
Yes in the future this is likely to change to the caller needing to know the store that contains their volume. In that case we will be given the storeID along with the device ID .